src/modules/utils/motordrivercontrol/MotorDriverControl.cpp

```diff
@@ -304,25 +328,27 @@ void MotorDriverControl::on_gcode_received(void *argument)
             // M911.3 S3 Zn setDoubleEdge Z=on|off Z1 is on Z0 is off
             // M911.3 S4 Zn setStepInterpolation Z=on|off Z1 is on Z0 is off
             // M911.3 S5 Zn setCoolStepEnabled Z=on|off Z1 is on Z0 is off
+			// M911.7 [p]hexdata[xN]  (Raw hexdata to/from SPI. See comments for WriteReadSPIstr function.)
 
-            if(gcode->subcode == 0 && gcode->get_num_args() == 0) {
+            if (gcode->subcode==7) { // Raw spi xmit and recv
```

Comment:
Not allowed, use a command.

Reply:
Jim:
I appreciate your comments and will be happy to do what makes the most sense for the project, but before that, I would like to poke at them a bit, constructively...

1. I thought that the "strict NIST RS274 compliance" train left the station a long time ago.  Isn't that getting a little orthodox at this point?

    - RepRap has several instances of strings in g-code line.  (I know Smoothieware g-code and RepRap g-code aren't meant to be totally compatible, but still it's a point.)  Perhaps I should add a letter ('S' or 'P') before the string and then put double quotes around the string, to make it a little more distinct and somewhat more compliant, as in the RepRap `M587`, etc.  I don't know if it really makes much more of a difference, though.

    - Smoothie's own (and RepRap's) `M28 <filename>` already "broke" NIST rules, and no quotes around this string either.

    - Smoothie's `M117 <panel_str>` is another case in point that has to be handled in such a special way that it had to be put into GcodeDispatch.cpp instead of where it truly belongs with the utils/panel code.  (I also note that this is another clear reason for providing access to `*command` in gcode.h, with respect to my comments to your other change request.  Providing `*command` access, however it is implemented [direct const char pointer access or accessor/iterator], actually allows Smoothieware to be more modular and self-consistent.)

    - I do not think it is in the Smoothieware contribution rules (i.e., strict NIST compliance), which is fine, but when did it become a thing?

    - What do you feel is really being broken, or what might this break, otherwise?

1. At first, I would have agreed that a simple_shell command would make sense here, but after thinking about it, it does not really work well for all cases, as it is intended.

    - On the one hand: a simple_shell command might make sense, if all of the drivers are daisy-chained together -- SDO of one tied to SDI of next, with a single common SSEL line to all of them (which is how I am currently configured).  This is one way to do it, such that one long write or read of the SPI bus writes/reads all chips at once (25 bytes every time in the case of 5x TMC5160 chips).  However, this is the only case that really makes sense.

    - On the other hand: the drivers could be configured such that all share a common SDI and a common SDO, with separate individual SSEL per chip.  (This is probably even the best way to configure them for future designs -- shorter write/read times, not forced to write/read all of them, which is actually more complex when all you wanted was to access just one.)  This configuration would (should) then require a simple_shell command to know which "axis" you wish to send the data to, which means it probably needs to have access to the MotorDriverControl module where it was assigned.  In which case, I assert that it is better off residing in the MotorDriverControl module where the axes and their individual SSEL lines are configured together -- again for proper modularity and consistency of Smoothieware.

    - In fact, the code that I added to the MotorDriverControl module is currently able to account for both SPI bus configurations -- daisy-chained, individual, or even a combination.  At least this is true for the configuration lines, as each `MotorDriverControl.{motor#}.reg#` line is written to the corresponding "motor#" chip-select.  _Having said that, I now realize that I should probably add an axis designator to the M911.7 gcode spec (if we keep it), so that it also properly handles both bus configurations._

1. The sendSPI function (short piece of code) that currently resides in MotorDriverConrol should really be moved out to lib, in any case.  (...and we would certainly want to if we moved my PR to simple_shell, otherwise we would need it in two places.)

Anyway, lots here, hopefully some good thoughtful philosophical arguments for you...

Best Regards,
Eric